udata/core/spatial/search.py

```diff
@@ -43,15 +46,18 @@ class Meta:
                               payloads=True)
 
     @classmethod
-    def compute_weight(cls, population):
-        """Weight must be in the interval [0..2147483647]"""
-        if 0 <= population <= 2147483647:
-            return population
-        else:
-            if population < 0:  # country/eh population is -99.
-                return 0
-            else:  # World population is 6772425850.
-                return 2147483647
+    def compute_weight(cls, zone):
+        '''
+        Give a weight to the zone according its administrative level first
+        and then its population.
+        '''
+        # Each level give give a step
```

Comment:
Done but I don't think it brings any value because this is relative scoring. What matter is that admin level weight more that population.